iris_grib/tests/integration/save_rules/test_grid_definition_section.py

```diff
@@ -0,0 +1,129 @@
+# (C) British Crown Copyright 2018, Met Office
+#
+# This file is part of iris-grib.
+#
+# iris-grib is free software: you can redistribute it and/or modify it under
+# the terms of the GNU Lesser General Public License as published by the
+# Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# iris-grib is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with iris-grib.  If not, see <http://www.gnu.org/licenses/>.
+"""
+Integration tests for :func:`iris_grib._save_rules.grid_definition_section`
+to confirm that the correct grid_definition_template is being selected.
+
+"""
+
+from __future__ import (absolute_import, division, print_function)
+from six.moves import (filter, input, map, range, zip)  # noqa
+
+# Import iris_grib.tests first so that some things can be initialised before
+# importing anything else.
+import iris_grib.tests as tests
+
+from iris.coords import DimCoord
+from iris.coord_systems import (GeogCS,
+                                RotatedGeogCS,
+                                TransverseMercator,
+                                LambertConformal,
+                                AlbersEqualArea)
+from iris.cube import Cube
+import numpy as np
+
+from iris_grib._save_rules import grid_definition_section
+from iris_grib.tests.unit.save_rules import GdtTestMixin
+
+
+class Test(tests.IrisGribTest, GdtTestMixin):
+    def setUp(self):
+        GdtTestMixin.setUp(self)
+        self.ellipsoid = GeogCS(6371200)
+
+    def _make_cube(self, x_points, y_points, cs, coord_units='1'):
```

Comment:
Could you not extend `GdtTestMixin._make_test_cube` to do this, instead of making a new thing ?
It looks like you want to duplicate that, but take control of the x/y names and units, so you could just add keywords for that.
(It already has coord-system control)

Reply:
@pp-mo yup - I had to add exactly this functionality in #116 anyhow, so I'll just make use of it here too.